src/app/http-interceptors/error.interceptor.spec.ts

```diff
@@ -0,0 +1,157 @@
+            (error:HttpErrorResponse) =>
+            {
+                expect(error.status).toEqual(401);
+                expect(error.error).toEqual(errorMsg);
+                expect(spy).toHaveBeenCalled();
+            }
+        );
+
+        const request:TestRequest = httpTestingController.expectOne(url);
+        request.flush(errorMsg, {status: 401, statusText: 'Error'});
+    });
+
+    it(`should catch error status 403 - Forbidden`, () =>
+    {
+        const errorMsg:string = 'Forbidden';
+        const url:string = '';
+        let spy:Spy = spyOn(alertService, 'error');
+
+        httpClient.get<Data>(url).subscribe(
+            () => fail('should have failed with the 403 error'),
+            (error:HttpErrorResponse) =>
+            {
+                expect(error.status).toEqual(403);
+                expect(error.error).toEqual(errorMsg);
+                expect(spy).toHaveBeenCalled();
+            }
+        );
+
+        const request:TestRequest = httpTestingController.expectOne(url);
+        request.flush(errorMsg, {status: 403, statusText: 'Error'});
+    });
+
+
+    it(`should log errors to the console in dev mode`, () =>
+    {
+        const errorMsg:string = 'Unauthenticated';
+        const url:string = '';
+
+        process.env.ENV = 'production';
+        let spy:Spy = spyOn(console, 'log');
+
+        httpClient.get<Data>(url).subscribe(
+            () => fail('should have failed with the 403 error'),
+            (error:HttpErrorResponse) =>
+            {
+                expect(error.status).toEqual(403);
+                expect(error.error).toEqual(errorMsg);
+                expect(spy).toHaveBeenCalledWith('status = '+403+ '\n' +
+                                                 'error = '+errorMsg);
+            }
+        );
+
+        const request:TestRequest = httpTestingController.expectOne(url);
+        request.flush(errorMsg, {status: 403, statusText: 'Error'});
+
+
+    });
+
+    it(`should log errors to the console if a support user is logged in`, () =>
+    {
+        pending();
+    });
+
+    it(`should pass on error response to be handled by the developer himself`, () =>
+    {
+        const errorMsg:string = 'Internal Server Error';
+        const url:string = '';
+
+        httpClient.get<Data>(url).subscribe(
+            () => fail('should have failed with the 500 error'),
+            (error:HttpErrorResponse) =>
+            {
+                expect(error.status).toEqual(500);
+                expect(error.error).toEqual(errorMsg);
+            }
+        );
+
+        const request:TestRequest = httpTestingController.expectOne(url);
+        request.flush(errorMsg, {status: 500, statusText: 'Error'});
+    });
+
+    it(`should not touch response if request is successful - status 2XX`, () =>
+    {
+        const url:string = '';
+
+        httpClient.get<Data>(url).subscribe(
+            (respone:Response) =>
+            {
+                expect(respone).toBeTruthy();
```

Comment:
response must be the same as you flushed..
it must be checked if it is the exact same object. And it must also be completely equal.